extensions/setname.md

```diff
@@ -0,0 +1,123 @@
+---
+title: IRCv3 `setname` Extension
+layout: spec
+copyrights:
+  -
+    name: "Janne Mareike Koschinski"
+    period: "2019"
+    email: "janne@kuschku.de"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `setname` capability name. Instead, implementations SHOULD use
+the `draft/setname` capability name to be interoperable with other software
+implementing a compatible work-in-progress version.
+
+The final version of the specification will use an unprefixed capability name.
+
+## Motivation
+
+Realnames have historically been immutable. This has never been much of an
+issue, as they were rarely used and usually not as present in client UIs.
+Still, multiple IRCds have provided independent, non-standard
+commands to update realnames.
+
+Nowadays, this has changed. Multiple clients show realnames directly in chat
+(e.g. next to the nickname) or use information from the realname to provide
+user-visible enhancements (e.g. using emails in the realname to provide
+gravatar based avatars).
+
+Due to this, the ability to change the realname has become relevant.
+
+## Description
+
+The `draft/setname` client capability allows clients to change their realname
+(GECOS). It also allows servers to directly inform clients about such a change.
+This avoids clients having to actually disconnect and reconnect. 
+
+This capability MUST be referred to as `draft/setname` at capability
+negotiation time.
+
+Servers advertising the `draft/setname` capability MUST implement the command
+as specified here, and MUST support usage of the command even while the
+capability is not negotiated.
+
+Servers advertising the capability MUST also provide a `NAMELEN` token in
+`RPL_ISUPPORT` with the maximum allowed length for realnames.
+
+If a client sends a `SETNAME` command without having negotiated the capability,
+the server SHOULD handle it silently, as the historic implementations did.
+
+Absence of this capability MUST NOT cause the client to prevent the user from
```

Comment:
This sentence effectively duplicates the one above:

> and MUST support usage of the command even while the
capability is not negotiated.

Reply:
What would you then suggest as alternative to replace these paragraphs:

> Servers advertising the `draft/setname` capability MUST implement the command as specified here, and MUST support usage of the command even while the capability is not negotiated.

> If a client sends a `SETNAME` command without having negotiated the capability, the server SHOULD handle it silently, as the historic implementations did.

Reply:
I'd honestly just leave this paragraph out, both of those paragraphs make different points but do explicitly say that the command is able to be used when the cap isn't negotiated.

Reply:
One of them says that the command MUST be usable, (but how servers should react to it is still unclear), the other says how they SHOULD react to it. Both are relevant, in a way.

Reply:
Yepyep. With the first one saying _"MUST support usage of the command even while the capability is not negotiated."_ makes this paragraph redundant. This doesn't add any information, state any new points, or state it in a more understandable way really (that initial line lays it out pretty well).

Reply:
so the main issue here is just the structure of the sentences was a bit confusing and presented a little backwards (as happens when we get different feedback as we push a spec through the process), made for some misreadings and made it unclear as to why the different sentences were included.

if we re-include text to this effect, a structure along these lines should be less confusing for readers:

>>>
Servers advertising the draft/setname capability MUST support usage of the command even while the capability is not negotiated. Clients MUST NOT prevent users from manually using the command while the capability is not negotiated.

If a client sends a SETNAME command without having negotiated the capability, the server SHOULD handle it silently (with no response), as historic implementations did.

Reply:
Sounds good to me.